lib/CL/clEnqueueNDRangeKernel.c

```diff
@@ -230,50 +228,25 @@ POname(clEnqueueNDRangeKernel)(cl_command_queue command_queue,
   POCL_RETURN_ERROR_COND((event_wait_list != NULL && num_events_in_wait_list == 0),
     CL_INVALID_EVENT_WAIT_LIST);
 
+  cache_lock = pocl_cache_acquire_writer_lock(kernel->program,
+                                              command_queue->device);
+  assert(cache_lock);
```

Comment:
By situation you mean "cache_lock == NULL" ? No, that doesn't happen in the scenario you described. Lockfiles contain the pid of the locking process, which is checked when you try to acquire the lock. So leftover lockfiles are not a problem. The NULL might happen if some process holds the lock for very long time, like minutes (so i think it shouldn't happen much).